### PR TITLE
fix half-constructed GOTO instruction

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/goto-programs/goto_convert.cpp [hunk 1]
```diff
@@ -109,28 +109,7 @@ void goto_convertt::finish_gotos(goto_programt &dest, const irep_idt &mode)
   {
     goto_programt::instructiont &i=*(g_it.first);
 
-    if(i.code.get_statement()=="non-deterministic-goto")
```
Comment (tautschnig) [inline src/goto-programs/goto_convert.cpp:112]: It may be worth mentioning this cleanup in the commit message.

**Review by tautschnig — APPROVED** [review]

**Comment by smowton** [comment]
The removal @tautschnig commented on is intentional?

**Comment by kroening** [comment]
Yes, we haven't had support for gotos with multiple targets for quite a while, i.e., this is dead code.

**Review by smowton — APPROVED** [review]
Suggest splitting remove-obsolete-feature and fix-half-constructed-instruction into separate commits, otherwise lgtm

**Comment by kroening** [comment]
all done

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: de57bb9bb65376812951459c6c30d8ae81350846).

**Review by peterschrammel — APPROVED** [review]
Fixing the tests should probably be squashed to preserve bisectability.

**Comment by tautschnig** [comment]
+1 for squashing for bisectability.

**Pull request merged**